### DISCUSSION TITLE
issues with creating a network on metamask

### DISCUSSION THREAD

**Discussion by workrate99** [Cyfrin/foundry-full-course-cu#3091] [Q&A]
after inputting the RPC url for anvil, when i put in the chain id it gives the error, "Could not fetch chain ID. Is your RPC URL correct? " 
![Screenshot (282)](https://github.com/user-attachments/assets/e37d1cef-bae7-483d-b9da-1e39683685ab)

**Comment by EngrPips** [comment]
Hello @workrate99, Please make sure your `Anvil` chain is actively running while you try to add it to metamask.

**Reply by workrate99** [reply]
yes the anvil chain is actively running
![Screenshot (283)](https://github.com/user-attachments/assets/ee9b52db-795b-4e6c-ac14-7509c98a29d4)

**Reply by EngrPips** [reply]
Why does it work for me and not for you?

![Screenshot 2024-12-23 at 11 47 57](https://github.com/user-attachments/assets/d28735b7-db4a-42bd-b3ed-2b514eb3654c)

![Screenshot 2024-12-23 at 11 47 39](https://github.com/user-attachments/assets/91c61683-5698-4a61-bc30-84d8ef233034)

**Reply by workrate99** [reply]
i don't really know or maybe its because i'm using gitpod, thats the only reason i can think of

**Reply by EngrPips** [reply]
I see. That could be the reason I am not running on Gitpod.

**Reply by satyaorz** [reply]
yeah seems like that, because the localhost on gitpod is not same as on your device.
can you try https://www.gitpod.io/docs/configure/workspaces/ports
idk, might work.

**Reply by EngrPips** [reply]
> yeah seems like that, because the localhost on gitpod is not same as on your device. can you try https://www.gitpod.io/docs/configure/workspaces/ports idk, might work.

Thanks for the heads-up Friend.
